### ISSUE TITLE
Unable to replicate results

### ISSUE THREAD

**Issue by Fairasname** [simula/mmsys2019-VISEM#1]
Hi,

I have started to work with your dataset, however, when I try to replicate the results of the frame experiments for motility prediction (progressive, non-progressive, and immotile), I obtain around half the error reported in your paper. For instance, using standard scikit-learn regressors, I obtain the following:

Standard ZeroRuleRegr (mean): CV training loss of 12.69 and validation loss of 13.16
Standard LinRegr:                        CV training loss of  7.45 and validation loss of 16.39
Standard RandForestReg:           CV training loss of  2.18 and validation loss of 13.48

Here is an example of the ZeroR for one fold.

 ```python
import numpy as np
import pandas as pd

from sklearn.dummy import DummyRegressor
from sklearn.model_selection import train_test_split

regressor = DummyRegressor()  # By default, it computes the mean of the labels
visem_df = pd.read_csv("visem.csv")[["Progressive motility (%)", 
                                     "Non progressive sperm motility (%)", 
                                     "Immotile sperm (%)"]]

X = np.random.rand(85*250)  # Dummy regressor does not care about X
y = []
for i in range(0, len(visem_df)):  # Repeat each output 250 times
    y.extend([visem_df.values[i].tolist()]*250) 
y = np.asarray(y)

# Assume possible splits are [(1 2), 0], [(0 2), 1], [(0 1), 2]
# 0 --> videos  1...28 (28 videos), frames     0... 6999, len is 7000
# 1 --> videos 29...56 (28 videos), frames  7000...13999, len is 7000
# 2 --> videos 57...85 (29 videos), frames 14000...21249, len is 7250
train_idx = list(range((1-1)*250, 28*250))  # We train/fit on the videos 1 to 28
train_idx.extend(list(range((57-1)*250, 85*250)))  # and 57 to 85 (2/3 of the data) 

valid_idx = list((range((29-1)*250, 56*250)))  # We validate on the videos 29 to 56 (1/3 of the data)

X_train, y_train = X[train_idx], y[train_idx]  # Get the training data
X_valid, y_valid = X[valid_idx], y[valid_idx]  # Get the validation data

regressor.fit(X_train, y_train)  # Fit on training data
train_loss = np.abs(regressor.predict(X_train) - y_train).mean()  # Train MAE
valid_loss = np.abs(regressor.predict(X_valid) - y_valid).mean()  # Valid MAE

# 12.849902534113061 12.58187134502924
print(train_loss, valid_loss)
```

Also, will the code of the experiments be released?

Thank you very much!

**Comment by kelkalot** [comment]
Hi,

Thanks a lot for the email. We used the Weka library version 3.8 for the
baseline experiments. We also performed cross validation.

Which features did you use for the experiments?
The plan is to release the code and features after we are back from
vacation (5th of August)

Best wishes,
Michael

lør. 13. jul. 2019 kl. 18:38 skrev Ivorra <notifications@github.com>:

**Comment by Fairasname** [comment]
Hi,

Thanks for the fast response. For the features, I am using 250 frames evenly extracted (using ffmpeg, jpg extension) from each video with 3-fold CV (validation splits are videos 1-28, 29-56, 57-85). For ZeroR, I have just tried on Weka and I am also obtaining similar results for ZeroR. 

Thanks again!

**Comment by kelkalot** [comment]
Hi,

Thanks for the details. We will release the code, features and arff files
ASAP we are back from vacation so that you can reproduce the results we
reported.

Best wishes,
Michael



lør. 13. jul. 2019 kl. 19:55 skrev Ivorra <notifications@github.com>:

**Comment by Fairasname** [comment]
Alright!

**Comment by kelkalot** [comment]
Hi,

I checked the results now in the paper. Reported is the additive error for
progressive, non-progressive and immobile which is exactly as reported in
the paper (38.68).
I will put the WEKA experiments and files used for it on the dataset page
this week! The error of 12.84 you get is the average I would assume (which
leads to an additive error of 38.55).

Best wishes,
Michael

**Comment by Fairasname** [comment]
I see, thanks for the clarification!
